jetty-server/src/main/java/org/eclipse/jetty/server/AbstractNCSARequestLog.java

```diff
@@ -489,4 +496,58 @@ public String getLogTimeZone()
     {
         return _logTimeZone;
     }
+    
+
+    private String extractUser(Request request)
```

Comment:
That is fine with me. I guess that leaves me with two options and I would like to ask which is preferable from your perspective so I do take the wrong one.

1) Use inheritance, as you suggested that would basically be to refactor the extract user to it's own method and make it protected. It would indeed be the least intrusive, but I think maybe it is abusing the inheritance mechanism.

2) Use dependency injection. Instead of using inheritance. One could create an interface UserExtractor, which is injected at construction time. We will need at least two constructors,
where one is the default constructor and a second one taking an instance implementing the UserExtractor interface. I will implement a class which does implement a class (AuthenticatedUserExtractor) which implement the UserExtractor interface, which is provided in case of the default constructor. 

I would prefer option two if I had to choose one of those two. What do you think? But I guess it would be a little bit slower than option 1, but maybe more focused on a single responsibility.